### PR TITLE
Allow specifying URLs for LaTeX engines

### PR DESCRIPTION
**Pull request by ianqsong**
add a field for MathJax CDN that user can input by calling Documenter.HTML

_Edit by @mortenpi: fix #1428_

### CODE REVIEW

**Comment by mortenpi** [comment]
Thanks for the PR!

I think it would be more natural to handle all of it in the `MathJax2/3` types. I.e. if you want to configure the URL, you'd just set something like `HTML(mathengine = MathJax3(url="https://...."))`. No reason to add another global to the `HTML` struct. Beyond that:

* Docstrings would need to be updated to reflect the new arguments.
* It would be good to add a test [to the example builds](https://github.com/JuliaDocs/Documenter.jl/blob/3b1934262da03c1bbf461420c46c9e12b5b05ec5/test/examples/make.jl#L196-L232) for overriding the URL, together with the relevant test [in the tests file](https://github.com/JuliaDocs/Documenter.jl/blob/master/test/examples/tests.jl) that would make sure that `documenter.js` contains the correct URL.

**Comment by ianqsong** [comment]
A new field `url` is added to `MathJax2/3`, of which the default value copies the original one, i.e., "cdnjs.cloudflare.com". So [the tests file](https://github.com/JuliaDocs/Documenter.jl/blob/master/test/examples/tests.jl) stays unchanged. 

As this is just very small enhancement, any maintainers could make it happen soon, right?

File: src/Writers/HTMLWriter.jl
Comment (mortenpi) [inline src/Writers/HTMLWriter.jl:534]: We need quotes here:

```suggestion
                script.src = '$url';
```

There is a `@test_broken` for this currently in the tests, so that would have to be updated as well.

**Review by mortenpi — CHANGES_REQUESTED** [review]
I took the liberty of modifying the tests a bit, to make sure we test all the cases. I would have two more requests before merging though:

1. Also add the same argument for `KaTeX` (for consistency).
2. Update the docstrings to reflect this argument

File: test/examples/make.jl
Comment (mortenpi) [inline test/examples/make.jl]: This build actually doesn't completely work (MathJax loads, but does not render the equation at the end of `man/tutorials.md`), but I am not sure why at the moment.

Comment (ianqsong) [reply test/examples/make.jl]: I don't have so much knowledge of MathJax2/3, but the config part in the example of MathJax2 above is the same as the one of MathJax3.
```
MathJax3(
    Dict(
        :loader => Dict("load" => ["[tex]/physics"]),
        :tex => Dict(
            "inlineMath" => [["\$","\$"], ["\\(","\\)"]],
            "tags" => "ams",
            "packages" => ["base", "ams", "autoload", "physics"],
        ),
    ),
 ),
```

**Comment by ianqsong** [comment]
> 1. Also add the same argument for `KaTeX` (for consistency).
I don't have much knowledge of `KaTax`, but there are three urls 
```
.../katex.min.css
.../katex.min.js
.../auto-render.min.js
```
Would you confirm that this is also the case for other CDNs?

I was thinking it be more efficient if I hand over this PR to you and you could do this in a single commit -- what'd you say?

**Comment by mortenpi** [comment]
> I was thinking it be more efficient if I hand over this PR to you and you could do this in a single commit -- what'd you say?

I'm afraid that might not really be the case -- it's much less time-consuming just to review, so it would probably be a while before I get around to this in that situation.

Regarding KaTeX: you are right, I had forgotten that it was a bit more complicated. Let's leave KaTeX as it is.

**Comment by ianqsong** [comment]
docstrings is updated. Many thanks for your help with the test cases.

**Review by mortenpi — APPROVED** [review]
Thanks @ianqsong! I took the liberty of doing a few edits. Once the tests pass, this should be good to go.

**Comment by mortenpi** [comment]
@ianqsong was the closing of the PR accidental?

**Comment by ianqsong** [comment]
sorry, a fat-finger mistake

**Pull request merged**
